modules/core/include/opencv2/core/ocl.hpp

```diff
@@ -276,6 +276,44 @@ class CV_EXPORTS Platform
     Impl* p;
 };
 
+/*
+// Attaches OpenCL context to OpenCV
+// Params:
+//   platformName - name of OpenCL platform to attach,
+//                  this string is used to check if platform is available
+//                  to OpenCV at runtime
+//   platfromID   - ID of platform attached context was created for
+//   context      - OpenCL context to be attached to OpenCV
+//   deviceID     - ID of device, must be created from attached context
+*/
+CV_EXPORTS void attachContext(const String& platformName, void* platformID, void* context, void* deviceID);
+
+/*
+// Convert OpenCL clBuffer to UMat
+// Note:
+//   OpenCL buffer (cl_mem_obj) should contain 2D image data, compatible with OpenCV.
+//   Memory content is not copied from clBuffer to UMat
+// Params:
+//   cl_mem_obj - source clBuffer handle
+//   step - num of bytes in single row
+//   rows - number of rows
+//   cols - number of cols
+//   type - OpenCV type of image
+//   dst  - destination UMat
+*/
+CV_EXPORTS void convertFromBuffer(void* cl_mem_obj, size_t step, int rows, int cols, int type, UMat& dst);
+
+/*
+// Convert OpenCL clImage to UMat
+// Note:
+//   OpenCL image (cl_mem_obj), compatible with OpenCV.
+//   Memory content is copied from clImage to UMat
+// Params:
+//   cl_mem_obj - source clImage handle
+//   dst        - destination UMat
+*/
+CV_EXPORTS void convertFromImage(void* cl_mem_obj, UMat& dst);
```

Comment:
cl_mem_obj --> cl_mem_buffer in `convertFromBuffer`
cl_mem_obj --> cl_mem_image in `convertFromImage`
?